src/librustc_codegen_llvm/abi.rs

```diff
@@ -507,6 +507,18 @@ impl<'tcx> FnTypeExt<'tcx> for FnType<'tcx, Ty<'tcx>> {
             }
         };
 
+        // Store the index of the last argument. This is useful
+        // for working with C compatible variadic arguments.
+        let last_arg_idx = if sig.inputs().len() > 0 {
+            Some(sig.inputs().len() - 1)
+        } else {
+            None
+        };
+
+        let va_list_did = match cx.tcx.lang_items().va_list() {
+            Some(did) => did,
+            None => bug!("va_list lang item required for C variadic functions"),
```

Comment:
This shouldn't bug when the function is not variadic. (Maybe move it to inside `if sig.variadic`.)

Reply:
Great catch!